tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -451,3 +451,42 @@ def test_diff_snapshot_overlay(guest_kernel, rootfs, microvm_factory):
     # Run some command to check that the restored VM works
     rc, _, stderr = new_vm.ssh.run("true")
     assert rc == 0, stderr
+
+
+def test_snapshot_overwrite_self(guest_kernel, rootfs, microvm_factory):
+    """Tests that if we try to take a snapshot that would overwrite the
+    very file from which the current VM is stored, nothing happens.
```

Comment:
does the file need to be of the memory size of the snapshotted VM or does it not matter?